### PR TITLE
Fix #12273: frozen columns right

### LINKED ISSUE #12273: DataTable: Align frozen columns right
### Description

Hello PrimeFaces Community,

I am proposing a new feature to enhance the PrimeFaces DataTable component by allowing the option to align frozen columns on the right side.

#### Key Points for Discussion:

- **Behavior**: How to determine which columns should be frozen.
- **Control**: How users can configure the alignment of frozen columns.

#### Current Progress:
I have successfully implemented a working demo for version 13.X and have written unit tests to ensure functionality. The implementation appears to be working well.

#### Next Steps:
I am considering opening a pull request. Before doing so, I would like to gather feedback on the following:

- The proposed behavior for determining and configuring frozen columns.
- Any additional considerations or potential improvements.

#### Control Mechanism:
To enable this feature, I have introduced a new attribute to the DataTable component: `alignFrozenColumnsRight`, which is a boolean value. When set to `true`, this attribute will align the frozen columns on the right side. When set to `false` (the default value), the columns will be aligned on the left side.

#### Behavior:

- **Left Aligned Frozen Columns:** The default behavior aligns the specified number of columns from the start of the column definition on the left side.
- **Right Aligned Frozen Columns:** With the `alignFrozenColumnsRight` attribute set to `true`, the specified number of columns will be aligned from the end of the column definition on the right side.

### PR DESCRIPTION
**Pull request by tvazquez-dev**
Fix #12273

### CODE REVIEW

File: primefaces/src/main/java/org/primefaces/component/datatable/DataTableBase.java [hunk 1]
```diff
@@ -67,6 +67,7 @@ public enum PropertyKeys {
         filterEvent,
         filteredValue,
         frozenColumns,
+        alignFrozenColumnsRight,
```
Comment (melloware) [inline primefaces/src/main/java/org/primefaces/component/datatable/DataTableBase.java]: rename this property to `frozenColumnsAlignment` and the values will be "right" or "left" with the default "left"

Then in the widget you can make a boolean like `var isFrozenColumnsRight = this.cfg.frozenColumnsAlignment === "right";`

**Review by melloware — CHANGES_REQUESTED** [review]

File: primefaces/src/main/java/org/primefaces/component/datatable/DataTableBase.java (hunk 1 above)
Comment (melloware) [reply primefaces/src/main/java/org/primefaces/component/datatable/DataTableBase.java]: Don't forget to update `/docs/15.0/whatsnew.md` and `/docs/15.0/components/datatable.md`

**Review by melloware — CHANGES_REQUESTED** [review]

File: primefaces/src/main/java/org/primefaces/component/datatable/DataTableRenderer.java [hunk 2]
```diff
@@ -432,7 +433,7 @@ protected void encodeScrollableTable(FacesContext context, DataTable table) thro
             writer.startElement("tbody", null);
             writer.startElement("tr", null);
 
-            if (table.isAlignFrozenColumnsRight()) {
+            if (table.getFrozenColumnsAlignment().equals("right")) {
```
Comment (melloware) [inline primefaces/src/main/java/org/primefaces/component/datatable/DataTableRenderer.java]:
```suggestion
            if ("right".equalsIgnoreCase(table.getFrozenColumnsAlignment())) {
```

**Review by melloware — CHANGES_REQUESTED** [review]

**Review by melloware — APPROVED** [review]

**Comment by Rapster** [comment]
Can't we do these tests in one IT instead of 2? They look really similar

**Comment by tandraschko** [comment]
+1 - please just check @Rapster comment

File: primefaces-integration-tests/src/main/webapp/datatable/dataTable046.xhtml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      xmlns:p="http://primefaces.org/ui">
+
+<f:view contentType="text/html;charset=UTF-8" encoding="UTF-8">
+    <h:head>
+
+    </h:head>
+
+    <h:body>
+
+        <h:form id="form">
```
Comment (melloware) [inline primefaces-integration-tests/src/main/webapp/datatable/dataTable046.xhtml:14]: As @Rapster mentioned combined these two tests into one test with a SelectBooleanButton you can just toggle in the test.

**Review by melloware — CHANGES_REQUESTED** [review]

**Review by melloware — APPROVED** [review]

**Comment by tandraschko** [comment]
13.x?

**Comment by melloware** [comment]
yep didn't realize this was against 13.X  should have noticed it. Going to merge to 15.0 main and revert this change.

**Pull request merged**